stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -0,0 +1,386 @@
+    {
+        std::vector<PointSpeedPair> points_and_target_speeds;
+        std::unordered_set<lanelet::Id> visited_lanelets;
+
+        bool first = true;
+        for(const auto& maneuver : maneuvers)
+        {
+            if(maneuver.type != cav_msgs::Maneuver::STOP_AND_WAIT)
+            {
+                throw std::invalid_argument ("Stop and Wait Maneuver Plugin doesn't support this maneuver type");
+            }
+            cav_msgs::StopAndWaitManeuver stop_and_wait_maneuver= maneuver.stop_and_wait_maneuver;
+
+            double starting_downtrack = stop_and_wait_maneuver.start_dist;  //starting downtrack recorded in message
+            if(first)   //check for first maneuver in vector 
+            {
+                if(starting_downtrack > max_starting_downtrack)
+                {
+                    starting_downtrack = max_starting_downtrack;
+                }
+                first = false;
+
+            }
+           
+            double ending_downtrack = stop_and_wait_maneuver.end_dist;
+            maneuver_time_ = ros::Duration(stop_and_wait_maneuver.end_time - stop_and_wait_maneuver.start_time).toSec();
+            
+            double delta_time, curr_time;
+            if(starting_downtrack == ending_downtrack)  //If at end_dist return zero speed trajectory
+            {
+                delta_time = min_timestep_;
+                curr_time = 0.0;
+                //wait
+                lanelet::BasicPoint2d curr_pose (pose_msg_.pose.position.x,pose_msg_.pose.position.y);
+                while(curr_time < minimal_trajectory_duration_)
+                {
+                    PointSpeedPair pair;
+                    pair.point = curr_pose;
+                    pair.speed = 0.0;
+                    points_and_target_speeds.push_back(pair);
+                    curr_time += delta_time;
+
+                    points_and_target_speeds.push_back(pair);
+                }
+            }
+            else
+            {
+                double jerk_req = (2*current_speed_)/pow(maneuver_time_,2);
+                double start_speed = current_speed_;    //Get static value of current speed at start of planning
+                if(jerk_req > max_jerk_limit_)
+                {
+                    //unsafe to stop at the required jerk - reset to max_jerk and go beyond the maneuver end_dist
+                    jerk_ = max_jerk_limit_;  
+                    maneuver_time_ = pow(2*start_speed, 0.5);
+                    double travel_dist_new = start_speed * maneuver_time_ - (0.167 * jerk_ * pow(maneuver_time_,3));
+                    ending_downtrack = travel_dist_new + starting_downtrack;
+                }
+                else jerk_ = jerk_req;
+
+                //get all the lanelets in between starting and ending downtrack on shortest path
+                auto lanelets = wm_->getLaneletsBetween(starting_downtrack, ending_downtrack, true);
```

Comment:
if ending downtrack has been extended beyond the value from the maneuver (because to stop at the originally planned ending downtrack would violate the jerk limit), and that stop is at the end of the route how does this behave? Downtrack is only defined for lanelets on the route. Something to ponder a bit as an edge case.

Reply:
It would still ignore the ending downtrack from the maneuver, in order to make sure the jerk limit is met.
Since there is no distinction between stop at the end of a route vs elsewhere, this check would have to happen in the strategic plugin

Reply:
I would probably recommend throwing some kind of exception, ROS_ERROR logging it, or something along those lines then, since we won't be able to access the route-lanelets for the part of the stop maneuver which extends beyond the end of the route.

Reply:
Following up on this, did we come to a decision on throwing an exception here or no?

Reply:
Yes we did, It is throwing an exceptions before this getLaneletsBetween call